### ISSUE TITLE
Support for multiplexed signals and enumeration types

### ISSUE THREAD

**Issue by 42niks** [jmagnuson/canparse#31]
@jmagnuson  Do you plan on implementing support for multiplexed signals and enumeration types?

**Comment by 42niks** [comment]
We are using canparse in our IoT Backend with a few automotive clients and the lack of enumeration and multiplexed signals is blocking us. I can give a shot at contributing this feature but need some direction from you in terms of overall design and what is required to get this done.

**Comment by wallem89** [comment]
I am also missing the enumeration type feature. Would be nice if you can add this @42niks or did you find another solution in Rust?

**Comment by 42niks** [comment]
Oh we are using python now :p there's a library cantools written in python and it has all features.

**Comment by wallem89** [comment]
@42niks Yes I know this one and using it for years. But I want to make something more embedded and learn more Rust on the way so therefore looking for a Rust alternative because I don't want to parse the DBC file myself

**Comment by jmagnuson** [comment]
These would be nice features to add-- they should be pretty compatible with how things are currently modeled, but unfortunately I don't have the bandwidth at the moment. I hope to have some free time for dev later in the summer, but would definitely expedite reviewing a PR.
